src/Symfony/Component/Translation/Bridge/Crowdin/Tests/CrowdinProviderTest.php

```diff
@@ -0,0 +1,709 @@
+                'headers' => [
+                    'Authorization' => 'Bearer API_TOKEN',
+                ],
+            ]), $this->getLoader(), $this->getLogger(), $this->getDefaultLocale(), 'api.crowdin.com/api/v2:99'),
+            'crowdin://api.crowdin.com/api/v2:99',
+        ];
+    }
+
+    public function testCompleteWriteProcessAddFiles()
+    {
+        $this->xliffFileDumper = new XliffFileDumper();
+
+        $getFileListResponse = $this->createMock(ResponseInterface::class);
+        $getFileListResponse->expects($this->any())
+            ->method('getContent')
+            ->with(false)
+            ->willReturn(json_encode(['data' => []]));
+
+        $addStorageResponse = $this->createMock(ResponseInterface::class);
+        $addStorageResponse->expects($this->any())
+            ->method('getStatusCode')
+            ->willReturn(201);
+        $addStorageResponse->expects($this->exactly(2))
+            ->method('getContent')
+            ->with(false)
+            ->willReturn(json_encode(['data' => ['id' => 19]]));
+
+        $addFileResponse = $this->createMock(ResponseInterface::class);
+        $addFileResponse->expects($this->any())
+            ->method('getStatusCode')
+            ->willReturn(201);
+        $addFileResponse->expects($this->exactly(2))
+            ->method('getContent')
+            ->with(false)
+            ->willReturnOnConsecutiveCalls(
+                json_encode(['data' => ['id' => 199, 'name' => 'messages.xlf']]),
+                json_encode(['data' => ['id' => 200, 'name' => 'validators.xlf']])
+            );
+
+        $expectedMessagesFileContent = <<<'XLIFF'
+<?xml version="1.0" encoding="utf-8"?>
+<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" version="1.2">
+  <file source-language="en" target-language="en" datatype="plaintext" original="file.ext">
+    <header>
+      <tool tool-id="symfony" tool-name="Symfony"/>
+    </header>
+    <body>
+      <trans-unit id="ypeBEso" resname="a">
+        <source>a</source>
+        <target>trans_en_a</target>
+      </trans-unit>
+    </body>
+  </file>
+</xliff>
+
+XLIFF;
+
+        $expectedValidatorsFileContent = <<<'XLIFF'
+<?xml version="1.0" encoding="utf-8"?>
+<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" version="1.2">
+  <file source-language="en" target-language="en" datatype="plaintext" original="file.ext">
+    <header>
+      <tool tool-id="symfony" tool-name="Symfony"/>
+    </header>
+    <body>
+      <trans-unit id="is7pld7" resname="post.num_comments">
+        <source>post.num_comments</source>
+        <target>{count, plural, one {# comment} other {# comments}}</target>
+      </trans-unit>
+    </body>
+  </file>
+</xliff>
+
+XLIFF;
+
+        $responses = [
+            'listFiles' => function (string $method, string $url, array $options = []) use ($getFileListResponse): ResponseInterface {
+                $this->assertEquals('GET', $method);
```

Comment:
You can use `assertSame` everywhere is possible (mainly for string and array comparisons)